### ISSUE TITLE
"no-default-export" and React.lazy

### ISSUE THREAD

**Issue by benny-medflyt** [palantir/tslint#4806]
I like the reasoning behind the [no-default-export](https://github.com/palantir/tslint/issues/1182) rule. But the React code-splitting feature (using `React.lazy`) seems to require the use of default exports. This is documented here:

<https://reactjs.org/docs/code-splitting.html#named-exports>

I was wondering if anyone else has any thoughts on this, or ways to still use named exports with React code splitting. Thank you

**Comment by JoshuaKGoldberg** [comment]
Hi @benny-medflyt! You can always disable the rule. Different codebases have different styles, and there's no single rule or ruleset that works for everyone.

#4305 tracks relaxing these stylistic preferences in the default ruleset.

**Comment by benny-medflyt** [comment]
Thanks for the response. I realize that this rule can be disabled.
I actually strongly believe in the reasoning behind the "no-default-export" rule, and agree that default exports should never be used. I opened this issue to check with the community if anyone has a recommendation for how to use "React.lazy" with named exports.

I looked at the code of "React.lazy" and I think it should be possible to re-implement it to so that it does not require default exports. I wonder if anyone has already done this.

Thank you
